autovizwidget/examples/autoviz.ipynb

```diff
@@ -0,0 +1,688 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# AutoVizWidget\n",
+    "\n",
+    "This is a library that allows you to visualize a [Pandas DataFrame](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.html) with a series of visualizations produced via [plotly](https://plot.ly/python/).\n",
+    "\n",
+    "The GUI displayed so that the user can easily select the features of the graph is produced via [ipywidgets](https://github.com/ipython/ipywidgets)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating a Pandas DataFrame\n",
+    "\n",
+    "We'll create a Pandas DataFrame to visualize."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "df = pd.DataFrame({'a': [1,2,3], 'b': [1,2,3], 'c': [\"hi\",\"hello\",\"hey\"]})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating an Encoding\n",
+    "\n",
+    "Encodings is how we indicate to the library which features it will use to create visualizations declaratively.\n",
+    "\n",
+    "Below, we'll select column `c` as the X axis, column `b` as the Y axis, and we'll take the `min` of the `b` column when plotting against the `c` column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from autovizwidget.widget.encoding import Encoding\n",
+    "\n",
+    "encoding = Encoding(chart_type=Encoding.chart_type_table, x='c', y='b', y_aggregation=Encoding.y_agg_min)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing\n",
+    "\n",
+    "We can now create an `AutoVizWidget` with our Pandas Dataframe (`df`) and the `encoding` we created."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from autovizwidget.widget.autovizwidget import AutoVizWidget\n",
+    "AutoVizWidget(df, encoding)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![widget2](Capture0.PNG)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Convenience methods\n",
+    "\n",
+    "Alternatively, you can let the library inspect your pandas DataFrame and select the X and Y axis automatically for you.\n",
```

Comment:
Something to consider (this is a nitpick) -- I might rather present `display_dataframe` as the default, standard way to display a dataframe, and then offer the `Encoding` method as a more low-level way to do this if you need more control.  I'm fine with either though.